docs/v0.14/api-plugin-base.txt

```diff
@@ -57,37 +60,130 @@ Base class has some methods to create configurable parameters for plugins, or to
+* `name`: parameter name as symbol
+* `type`: parameter type
+* `options`: options for parameter
+* `block`: if block is given, convert value via given block
+
+About types and options, see LINK:[Types of Configuration Parameters](api-config-types).
+
+    :::ruby
+    config_param :name, :string, default: "John Doe", alias: :full_name
+
+    config_param :pattern, do |value|
+      Regexp.compile(value)
+    end
+
+    config_param :delimiter, default: "\t" do |value|
+      case value
+      when /SPACE/i then " "
+      when /COMMA/i then ","
+      else "\t"
+      end
+    end
 
 #### .config_set_default(name, default_value)
 
-TODO: write this section later
+Set default value to `name`. If the default value has already been existed, raise `ArgumentError`.
+
+* `name`: The name of parameter
+* `default_value`: Default value for the parameter
+
+    :::ruby
+    # lib/fluent/plugin/buffer.rb
+    config_param :chunk_limit_size, :size, default: DEFAULT_CHUNK_LIMIT_SIZE
+
+    # lib/fluent/plugin/buf_file.rb
+    config_set_default :chunk_limit_size, DEFAULT_CHUNK_LIMIT_SIZE
 
 #### .config_set_desc(name, description)
 
-TODO: write this section later
+Set description to `name`. If the description has already been existed, raise `ArgumentError`.
+This is for internal API. Use `desc` instead.
+
+* `name`: The name of parameter
+* `description`: Description about the parameter
 
 #### .desc(description)
 
-TODO: write this section later
+Set description to immediately following parameter.
+
+* `description`: Description about the parameter
+
+    :::ruby
+    desc "description about following parameter"
+    config_param :user, :string
 
 #### .config_section(name, **options, &block)
 
-TODO: write this section later
+Define a section to construct structured configuration.
+
+* `name`: The name of section
+* `options`
+  * `root`: If true, this section is root section. This is only for internal use.
+  * `param_name`: This section name
+  * `final`: If true, subclass of this class cannot overwrite this section. e.g. 3rd party plugins cannot overwrite buffer section
+  * `init`: If true, parameters in this section must have default value. 3rd party plugins don't have to be conscious about this option.
+  * `required`: If true, this section is required. Fluentd will raise `Fluent::ConfigError` when this section is missing in configuration.
+  * `multi`: If true, users can write this section in configuration.
+  * `alias`: Alias of this section
+
+Code Example:
+
+    :::ruby
+    config_section :user, multi: true do
+      config_param :name, :string
+      config_param :password, :string, secret: true
+    end
+    
+    # nested sections
+    config_section :child, :param_name: "children", required: false, multi: true do
+      config_param :name, :string
+      config_param :power, :integer, default: nil
+      config_section :item, multi: true do
+        config_param :name, :string
+        config_param :flavor, :string
+      end
+    end
```

Comment:
Could you add actual configuraton for it?

```
<user>
  name xxx
  password yyy
</user>
```

and

```
<child>
  name xxx
  <item>
    name xxx
    flavior yyy
  </item>
```

Reply:
Sure!